src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -20,6 +20,7 @@ internal sealed class SourceGenJsonTypeInfo<T> : JsonTypeInfo<T>
         public SourceGenJsonTypeInfo(JsonConverter converter, JsonSerializerOptions options)
             : base(converter, options)
         {
+            PolymorphismOptions = JsonPolymorphismOptions.CreateFromAttributeDeclarations(Type);
```

Comment:
won't this use reflection for source gen? should we possibly make source gen set this when constructing JsonTypeInfo?

Reply:
It will. Evading reflection would require exposing ad hoc APIs in `JsonObjectInfoValues<T>` and `JsonObjectInfoValues<T>`. I don't think it's worth the extra effort given that reading attributes is linker-safe. If we do introduce breaks in the `JsonMetadataServices` infrastructure for sourcegen vNext I think it should simply target JsonTypeInfo APIs directly.

Reply:
It won't because you've just exposed the APIs in this PR so you can just modify JsonTypeInfo directly. Feel free to ignore this comment though - as we discussed offline it doesn't seem it will have any linker impact

Reply:
Yes but the source generator does not interface with JsonTypeInfo directly when populating metadata.